code/game/objects/items/weapons/his_grace.dm

```diff
@@ -204,3 +208,26 @@
 			if(prev_bloodthirst >= HIS_GRACE_PECKISH)
 				master.visible_message("<span class='warning'>[src] is satiated.</span>", "<span class='his_grace big'>[src]'s hunger recedes...</span>")
 	force = initial(force) + force_bonus
+
+/obj/item/weapon/his_grace/proc/ascend()
+	var/obj/item/weapon/his_grace/gold/G = new
```

Comment:
You should probably just set the ascended var and change the name and sprite instead of making a new object.